pkg/bgpv1/manager/manager.go

```diff
@@ -108,7 +109,10 @@ func NewBGPRouterManager(params bgpRouterManagerParams) agent.BGPRouterManager {
 //
 // ConfigurePeers should return only once a subsequent invocation is safe.
 // This method is not thread safe and does not intend to be called concurrently.
-func (m *BGPRouterManager) ConfigurePeers(ctx context.Context, policy *v2alpha1api.CiliumBGPPeeringPolicy, node *node.LocalNode) error {
+func (m *BGPRouterManager) ConfigurePeers(ctx context.Context,
+	policy *v2alpha1api.CiliumBGPPeeringPolicy,
+	node *node.LocalNode,
+	ciliumNode *v2api.CiliumNode) error {
```

Comment:
nit (feel free to ignore): Would it make sense to introduce a new `localNode` struct that contains both the `node.LocalNode` as well as the `CiliumNode`? It feels a bit weird that we're basically passing in the same node object twice.

Reply:
Might even be worth adding a reference local cilium node to the LocalNode type in the LocalNodeStore - @jussi wdyt?

Reply:
> nit (feel free to ignore): Would it make sense to introduce a new localNode struct that contains both the node.LocalNode as well as the CiliumNode? It feels a bit weird that we're basically passing in the same node object twice.

I agree but the [plan](https://github.com/cilium/cilium/pull/27100#issuecomment-1707611384) is to remove the `node.LocalNode` after this PR merges.